server/dialect-idms/src/main/antlr4/org/eclipse/lsp/cobol/dialects/idms/IdmsParser.g4

```diff
@@ -1191,3 +1191,21 @@ cobolCompilerDirectivesKeywords
 endClause
     : (DOT_FS | SEMICOLON_FS)
     ;
+// --------- TOLERATE LR STATEMENTS ---------------//
+idmsLRStatements: obtainLRStatement;
+// obtain LR statement toleration
+obtainLRStatement
+    : OBTAIN (FIRST|NEXT)? logicalRecordName
+      (INTO altLogicalRecordLocation)?
+      (WHERE booleanExpression)?
+      imperativeStatementCall?
+    ;
+pathStatus: {validateLength(_input.LT(1).getText(), "path-status", 16);} cobolWord;
+imperativeStatementCall: ON pathStatus imperativeStatement;
+imperativeStatement: (~DOT_FS)*;
```

Comment:
Wouldn't it be sufficient to just match the part we intend to replace?
```suggestion
imperativeStatementCall: ON pathStatus;
```
Does the IDMS statement actually have to end with `.`? If not, we might match the whole sentence instead of just a statement.

Reply:
updated